src/generators/java/presets/CommonPreset.ts

```diff
@@ -110,16 +199,24 @@ export const JAVA_COMMON_PRESET: JavaPreset = {
       const shouldContainEqual = options.equal === undefined || options.equal === true;
       const shouldContainHashCode = options.hashCode === undefined || options.hashCode === true;
       const shouldContainToString = options.classToString === undefined || options.classToString === true;
+      const shouldContainMarshal = options.marshal === undefined || options.marshal === true;
 
-      if (shouldContainEqual === true || shouldContainHashCode === true) {
+      if (shouldContainEqual === true || shouldContainHashCode === true || shouldContainMarshal === true) {
         renderer.addDependency('import java.util.Objects;');
+        renderer.addDependency('import java.util.stream;');
```

Comment:
You are missing a `Map` import 🙂

Reply:
Also please make a new if statement for adding dependencies for `shouldContainMarshal`. As you do not want these imports present if `shouldContainMarshal == false`.